x-pack/legacy/plugins/ml/public/application/jobs/jobs_list/components/ml_job_editor/ml_job_editor.tsx

```diff
@@ -4,23 +4,33 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import PropTypes from 'prop-types';
-import React from 'react';
+import React, { FC } from 'react';
 
 import { EuiCodeEditor } from '@elastic/eui';
+import { xJsonMode } from '../../../../components/custom_hooks';
 
-export const EDITOR_MODE = { TEXT: 'text', JSON: 'json' };
+export const ML_EDITOR_MODE = { TEXT: 'text', JSON: 'json', XJSON: xJsonMode };
```

Comment:
I wonder if this should remain caps-snake-case. In the previous version this was a plain JS-style enum, now we're in TypeScript land and it contains a non-enum like attribute. I don't have strong opinions about it but it feels a bit strange with the mix of attribute types.